pymc3/tuning/starting.py

```diff
@@ -171,3 +174,134 @@ def allinmodel(vars, model):
     notin = [v for v in vars if v not in model.vars]
     if notin:
         raise ValueError("Some variables not in the model: " + str(notin))
+
+
+
+class Monitor(object):
```

Comment:
Gosh, not I've seen.  Couldn't find anything similar after a quick look around.  There certainly might be, I will look around.  I took the idea from GPy, theirs was made from scratch also.